cln_lspd.go

```diff
@@ -0,0 +1,319 @@
+	}
+
+	if paymentSecret != nil {
+		if fundingTxID == nil {
+			if hex.EncodeToString(paymentHash) == event.Htlc.PaymentHash {
+
+				fundingTxID, fundingTxOutnum, err = clnOpenChannel(clientcln, hex.EncodeToString(paymentHash), hex.EncodeToString(destination), incomingAmountMsat)
+				log.Printf("openclnOpenChannelChannel(%v, %v) err: %v", destination, incomingAmountMsat, err)
+				if err != nil {
+					log.Printf(" clnOpenChannel error: %v", err)
+				}
+
+			} else { //probing
+				failureCode := "TEMPORARY_CHANNEL_FAILURE"
+				if err = clnIsConnected(clientcln, hex.EncodeToString(destination)); err == nil {
+					failureCode = "INCORRECT_OR_UNKNOWN_PAYMENT_DETAILS"
+				}
+				log.Printf(" failureCode %v , Error %v", failureCode, err)
+				return event.Continue(), err
+			}
+
+		}
+		var h chainhash.Hash
+		err = h.SetBytes(fundingTxID)
+		if err != nil {
+			log.Printf("h.SetBytes(%v) error: %v", fundingTxID, err)
+		}
+		channelPoint := wire.NewOutPoint(&h, fundingTxOutnum).String()
+		return clnResumeOrCancel(clientcln, hex.EncodeToString(destination), hex.EncodeToString(fundingTxID), hex.EncodeToString(paymentHash), uint64(outgoingAmountMsat), 99, event, channelPoint)
+
+	}
+
+	return event.Continue(), err
+}
+
+func clnGetChannel(clientcln *glightning.Lightning, destination string, fundingTxID string) string {
+	obj, err := clientcln.ListFunds()
+	if err != nil {
+		log.Printf("clientcln.ListChannels(%v) error: %v", destination, err)
+		return ""
+	}
+	for _, c := range obj.Channels {
+		log.Printf("getChannel(%v): %v", destination, c.ShortChannelId)
+		if c.FundingTxId == fundingTxID && c.Connected {
+			chanid := c.ShortChannelId
+			return chanid
+		}
+	}
+	log.Printf("No channel found: getChannel(%v)", destination)
+	return ""
+}
+func clnResumeOrCancel(clientcln *glightning.Lightning, destination string, fundingTxID string, paymentHash string, outgoingAmountMsat uint64, riskfactor float32, event *glightning.HtlcAcceptedEvent, channelPoint string) (*glightning.HtlcAcceptedResponse, error) {
+	deadline := time.Now().Add(60 * time.Second)
+
+	for {
+		shortChanID := clnGetChannel(clientcln, destination, fundingTxID)
+
+		if shortChanID != "" {
+			log.Printf("channel opended successfully chanID: %v)", shortChanID)
+			dest, err := hex.DecodeString(destination)
+			if err != nil {
+				log.Printf("hex.DecodeString(destination) error: %v", err)
+				break
+			}
+			fields := strings.Split(shortChanID, "x")
+			var chanID lnwire.ShortChannelID
+			var temp []int64
+			for i, field := range fields {
+				temp[i], _ = strconv.ParseInt(field, 10, 64)
+			}
+			chanID = lnwire.ShortChannelID{BlockHeight: uint32(temp[0]), TxIndex: uint32(temp[1]), TxPosition: uint16(temp[2])}
+			// we don't get short channel id until 1 conf
+			err = insertChannel(chanID.ToUint64(), channelPoint, dest, time.Now())
+			if err != nil {
+				log.Printf("insertChannel error: %v", err)
+			}
+
```

Comment:
log.Printf("forwarding htlc to the destination node and a new private channel was opened")
return event.Continue()